public/app/features/dashboard-scene/saving/SaveDashboardAsForm.tsx

```diff
@@ -200,3 +202,12 @@ async function validateDashboardName(title: string, formValues: SaveDashboardAsF
     return e instanceof Error ? e.message : 'Dashboard name is invalid';
   }
 }
+
+function getDashboardIdFromScene(dashboard: DashboardScene) {
```

Comment:
I had to create this function to get the dashboard `id` deprecated prop from dashboard scene meta. I assumed if we have `k8s`, is because we are dealing with k8s dashboards. Is there a better approach to fix this issue? 

cc @dprokop @ivanortegaalba @harisrozajac

Reply:
you can simply set `dashboard.state.id` from resource's `metadata.labels` when creating the scene. Id retrieval should not depend on the response implementation.

Reply:
you are already setting it here: https://github.com/grafana/grafana/pull/99590/files#diff-89fa6a2d879cff717639a3f4ebcfc8b5939688fc89987c0c4bd3c8bac9a76c89R167

Reply:
Oh good catch, let me try that